store/layer_generators/main.go

```diff
@@ -92,10 +115,20 @@ func ExtractStoreMetadata() StoreMetadata {
 					if _, ok := topLevelFunctions[methodName]; ok {
 						params := []Param{}
 						results := []string{}
-
+						paramsToTrace := map[string]bool{}
 						ast.Inspect(method.Type, func(expr ast.Node) bool {
 							switch e := expr.(type) {
 							case *ast.FuncType:
+								if method.Doc != nil {
+									for _, comment := range method.Doc.List {
+										s := comment.Text
+										if idx := strings.Index(s, OPEN_TRACING_PARAMS_MARKER); idx != -1 {
+											for _, p := range strings.Split(s[idx+len(OPEN_TRACING_PARAMS_MARKER):], ",") {
+												paramsToTrace[strings.TrimSpace(p)] = true
+											}
+										}
+									}
+								}
 								if e.Params != nil {
 									for _, param := range e.Params.List {
```

Comment:
I was thinking of an edge-case. What if someone changes a param name and then forgets to update the opentracing param comment ? Can we add a check here after collecting the `paramsToTrace` and `params`, where we just iterate `paramsToTrace` and confirm that every element is present in `params` ? If not, then raise an error.